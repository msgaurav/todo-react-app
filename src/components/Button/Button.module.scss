#Button_wrapper {
    min-width: 100px;
    font-size: 1.2rem;
    background-color: steelblue;
    color: white;
    border: none;
    border-radius: 3px;
    padding: 7px;
    margin-left: 0.5rem;
    cursor: pointer;
    box-shadow: inset 2px 2px 3px rgba(255, 255, 255, .6),
    inset -2px -2px 3px rgba(0, 0, 0, .6);

    &:hover {
        background-color: #1a71ba;
    }

    &:active {
        box-shadow: inset -2px -2px 3px rgba(255, 255, 255, .6),
        inset 2px 2px 3px rgba(0, 0, 0, .6);
    }
}