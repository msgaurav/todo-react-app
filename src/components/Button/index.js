/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import PropTypes from 'prop-types'
import styles from './Button.module.scss'

export default function Button(props) {
  const { title, ...rest } = props

  return (
    <button
      id={styles.Button_wrapper}
      type="button"
      {...rest}
    >
      {title}
    </button>
  )
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
}
