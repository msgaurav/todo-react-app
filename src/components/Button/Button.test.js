import React from 'react'
import { render } from '@testing-library/react'
import Button from './index'

const propsData = {
  title: 'Submit',
}

describe('The Button Component', () => {
  it('snapshot renders correctly', () => {
    const { asFragment } = render(
      <Button
        title={propsData.title}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should renders with correct title', () => {
    const { queryByText } = render(
      <Button
        title={propsData.title}
      />,
    )
    expect(queryByText('Submit')).toBeTruthy()
  })
})
