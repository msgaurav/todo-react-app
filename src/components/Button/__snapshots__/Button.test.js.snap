// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button Component snapshot renders correctly 1`] = `
<DocumentFragment>
  <button
    id="Button_wrapper"
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;
